internal/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,162 @@
+	return nil
+}
+
+func (np *networkPolicy) ensureOwnerReference(ctx context.Context, policy *networkingv1.NetworkPolicy, owner metav1.Object) error {
+	logger := log.FromContext(ctx)
+
+	for _, existingOwner := range policy.GetOwnerReferences() {
+		if existingOwner.UID == owner.GetUID() {
+			logger.V(1).Info("Owner already exists on NetworkPolicy", "policy", policy.Name, "owner", owner.GetName())
+			return nil
+		}
+	}
+
+	logger.V(1).Info("Adding owner reference to existing NetworkPolicy", "policy", policy.Name, "owner", owner.GetName())
+	existingCopy := policy.DeepCopy()
+	if err := controllerutil.SetOwnerReference(owner, policy, np.scheme); err != nil {
+		return err
+	}
+
+	if err := np.client.Patch(ctx, policy, client.MergeFrom(existingCopy)); err != nil {
+		return fmt.Errorf("failed to update NetworkPolicy owner references %s/%s: %v", policy.Namespace, policy.Name, err)
+	}
+
+	logger.Info("Successfully updated NetworkPolicy owner references", "name", policy.Name, "namespace", policy.Namespace, "owner", owner.GetName())
+	return nil
+}
+
+func (np *networkPolicy) WorkerPodNetworkPolicy(namespace string) *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workerPodNetworkPolicyName,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/component": "worker",
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	}
+}
+
+func (np *networkPolicy) BuildSignPodsNetworkPolicy(namespace string) *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      buildSignPodsNetworkPolicyName,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "openshift.io/build.name",
+						Operator: metav1.LabelSelectorOpExists,
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{}, // Allow all egress
+			},
+		},
+	}
+}
+
+func (np *networkPolicy) PullPodNetworkPolicy(namespace string) *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pullPodNetworkPolicyName,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      pod.PullPodTypeLabelKey,
+						Operator: metav1.LabelSelectorOpExists,
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
```

Comment:
Is this a deny-all policy? (I deduced it from the worker policy). We only need incoming traffic isn't it?